### PR TITLE
[netcore] Smarter version numbering of nupkg

### PR DESCRIPTION
**Pull request by directhex**
`nuget pack runtime.nuspec -properties VERSION=6.3.0.208\;RID=linux-x64\;PLATFORM_AOT_SUFFIX=.so\;COREARCH=x64`

### CODE REVIEW

File: netcore/Makefile.am [hunk 1]
```diff
@@ -57,7 +57,9 @@ link-mono:
 prepare: $(NETCORESDK_FILE) update-corefx link-mono
 
 nupkg:
-	nuget pack runtime.nuspec -properties VERSION=@VERSION@\;RID=@RID@\;PLATFORM_AOT_SUFFIX=@PLATFORM_AOT_SUFFIX@\;COREARCH=@COREARCH@
+BLAME = $(shell git blame $(top_srcdir)/configure.ac | grep AC_INIT | cut -f1 -d' ')
+VERSTUB = .$(shell git log $(BLAME)..HEAD --oneline | wc -l)
```
Comment (filipnavara) [inline netcore/Makefile.am]: nit: `git rev-list --count $(BLAME)..HEAD`

I know there are other places in the build scripts doing this, so it's probably worth a separated patch just for that.

Comment (akoeplinger) [reply netcore/Makefile.am]: Interesting, I've never seen variables being defined inside of a Makefile target. Is that an automake feature?

Comment (filipnavara) [reply netcore/Makefile.am]: I do it in regular Makefiles, so I think it's quite universal. I never seem to remember the difference between `:=` and `=` though.

**Comment by directhex** [comment]
You say that, and now I'm worried it's not doing the right thing, i.e. always evaluating the variables. I'll test more on Monday to ensure it's working as planned.

**Comment by directhex** [comment]
`$(top_srcdir)` isn't populated without .am :roll_eyes: looking for an adequate replacement. Maybe just ..

**Comment by directhex** [comment]
OK. This is working as expected in VSTS.

**Comment by directhex** [comment]
@monojenkins build failed

**Comment by directhex** [comment]
@monojenkins build failed

**Comment by EgorBo** [comment]
Btw, the content of this nuget package is wrong
1) System.Private.CoreLib.dll should be under `runtimes/osx-x64/lib/netcoreapp3.0`
2) `libmonosgen-2.0.dylib` for now should be `libcoreclr.dylib` in the output dir

**Comment by directhex** [comment]
@EgorBo We are not pretending to be CoreCLR in what we deliver to the Core team. It's on them to cleanly consume Mono. We need to be in a position to give them something they can start integrating.

**Comment by EgorBo** [comment]
@directhex ah, ok

**Comment by directhex** [comment]
@EgorBo I took the directory layout in the nupkg from e.g. runtime.rhel.6-x64.Microsoft.NETCore.Runtime.CoreCLR

**Pull request merged**